LiskHQ/lisk-sdk
# Updated signature and heightPrevious / heightPrevoted with support old implementation - Closes #3558, #3556

### What was the problem?
Addresses both the changes needed for heightPrevious and heightPrevoted properties, and supports both old an new implementation based upon `bftUpgradeHeight` property which can be set to tell the application to switch to the new block headers with heightPrevious/prevoted included.
--> Uses migration to alter when upgrading, does not affect old behavior.

As the `getBytes()` function is needed for this change, it's included in the PR. It adds the `height`, `heightPrevious`, and `heightPrevoted` to the signature.

_Edit: Property names changed to: `maxHeightPreviouslyForged` and `prevotedConfirmedUptoHeight`._

### How to test it?
Run network, check for both new block header properties to be present in the database for `blocks`.

### Review checklist

* The PR resolves #3556, #3558 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated
